services/has-many-getter.js

```diff
@@ -111,7 +112,17 @@ function HasManyGetter(model, association, opts, params) {
   this.perform = function () {
     return getRecords()
       .then(function (records) {
-        return [count, records];
+        var decorators = null;
+
+        if (params.search) {
+          decorators = RecordsDecorator.decorateForSearch(
+            getRecords(),
+            searchBuilder.getFieldsSearched(),
+            params.search
+          );
+        }
+
+        return [records, count, decorators];
```

Comment:
I think you didn't test this code:

<img width="999" alt="screen shot 2018-05-22 at 13 51 10" src="https://user-images.githubusercontent.com/664261/40360689-39c83e06-5dc7-11e8-95dd-ff766a3cba8e.png">